User Guides/1-Prerequisite Steps Before Deployment.md

```diff
@@ -39,8 +39,8 @@ $childCert = Get-ChildItem -Path cert:\CurrentUser\My | ?{ $_.Subject -eq "CN=Co
 $type = [System.Security.Cryptography.X509Certificates.X509Certificate]::pfx
 $securePassword = ConvertTo-SecureString -String "MyPassword" -Force –AsPlainText
 
-Export-PfxCertificate -Cert $rootCert -FilePath "ContosoRootCertificate.pfx" -Password $securePassword -Verbose
-Export-PfxCertificate -Cert $childCert -FilePath "ContosoChildCertificate.pfx" -Password $securePassword -Verbose
+Export-PfxCertificate -Cert $rootCert[1] -FilePath "ContosoRootCertificate.pfx" -Password $securePassword -Verbose
```

Comment:
This index reference assumes the certificate is the last cert in an array of count 2. This only true for your machine. It appears you have two certificates of name "CN=ContosoRootCertificate". Presumably you ran DeployVPN.ps1 twice? To accept a fix for this, you'll need to check $rootCert for a count greater than zero and either throw an error or use something similar to "Select -First 1".

Reply:
Great catch. You are absolutely right. I ran DeployVPN.ps1 2 times due to a network issue.
I can check both the arrays for count and select first value if count is greater than 1.

```
If($rootCert.Count -gt 1) {
 Echo "More than one certificate by the name ContosoRootCertificate, selecting first certificate"
 $rootCert = $rootCert[0] #Selecting first certificate
}
```

```
If($childCert.Count -gt 1) {
 Echo "More than one certificate by the name ContosoChildCertificate, selecting first certificate"
 $childCert = $childCert[0] #Selecting first certificate
}
```
I also didn't see any use of variable **$type** in the script. It can be removed or commented ?
Thanks for your time and feedback !

Reply:
The conditionals you described look good. Just use Write-Output instead of Echo.
As for $type, feel free to remove.